src/progress/Progress.ts

```diff
@@ -0,0 +1,54 @@
+import { WidgetBase } from '@dojo/widget-core/WidgetBase';
+import { DNode } from '@dojo/widget-core/interfaces';
+import { ThemedMixin, ThemedProperties, theme } from '@dojo/widget-core/mixins/Themed';
+import { v } from '@dojo/widget-core/d';
+import * as css from './styles/progress.m.css';
+
+/**
+ * @type ProgressProperties
+ *
+ * Properties that can be set on a Progress component
+ *
+ * @property value            The current value
+ * @property output...........A function used to determine the output display
+ * @property showOutput.......Toggles visibility of progess bar output
+ * @property max..............Value used to calculate percent width
+ */
+export interface ProgressProperties extends ThemedProperties {
+	value: number;
+	output?(value: number, percent: number): string;
+	showOutput?: boolean;
+	max?: number;
+}
+
+export const ProgressBase = ThemedMixin(WidgetBase);
+
+@theme(css)
+export default class Progress extends ProgressBase<ProgressProperties> {
+	private _output(value: number, percent: number) {
+		const { output } = this.properties;
+		return output ? output(value, percent) : `${percent}%`;
+	}
+
+	render(): DNode {
+		const {
+			value,
+			showOutput = true,
+			max = 100
+		} = this.properties;
+
+		const percent = Math.round((value / max) * 100);
+
+		return v('div', { classes: this.theme(css.root) }, [
+			v('div', { classes: this.theme(css.bar) }, [
```

Comment:
Additional properties on the bar:
```
role: 'progressbar',
'aria-valuemin': `${min}`,
'aria-valuemax': `${max}`,
'aria-valuenow': `${value}`,
'aria-valuetext`: output ? `${output(value, percent)}` : null
```

There are some other attributes that would need to be added and controlled if the `aria-*` thing falls through, like `aria-labelledby`, `aria-live` (probably also aria-atomic). It's possible we should add them now, so we don't forget and so it isn't lacking in the meantime.

Reply:
Also the bit with `aria-valuetext` doesn't draw from `this._output` because it should only be present if custom text is supplied

Reply:
Thanks for these

Reply:
re. `aria-valuetext`, i think that it *should* draw from the `this._output` function because it will at least always add `%` to the value so it will be different to the value itself.

Reply:
I think the `aria-valuetext` thing should be OK with `this._output`. Screenreaders do actually add "percent" when no value text is given, and I usually tend towards adding the least aria possible. On the other hand, always using `this._output` is cleaner and it should be the same experience. I'll double check w/ NVDA and if that reads fine then I'm good leaving off the extra conditional.